vnext/ReactUWP/Modules/Animated/NativeAnimatedModule.h

```diff
@@ -0,0 +1,104 @@
+  /// whenever you render Animated.View and stores a mapping from the view
+  /// props to the corresponding animated values (so it's actually also
+  /// a node with connections to the value nodes).
+  /// 
+  /// Last "special" elements of the the graph are "animation drivers". Those
+  /// are objects (represented as a graph nodes too) that based on some
+  /// criteria updates attached values via a composition expression animation
+  /// (we have few types of those, e.g., spring, timing, decay). Animation objects
+  /// can be "started" and "stopped". Those are like "pulse generators" for the
+  /// rest of the nodes graph. Those pulses then propagate along the graph to the
+  /// children nodes up to the special node type: AnimatedProps which then 
+  /// can be used to calculate prop update map for a view.
+  /// 
+  /// This class acts as a proxy between the "native" API that can be called
+  /// from JS and the main class that coordinates all the action: 
+  /// <see cref="NativeAnimatedNodeManager"/>.
+  /// </remarks>
+namespace react { namespace uwp {
+  class NativeAnimatedModule final : public facebook::xplat::module::CxxModule
+  {
+  public:
+    NativeAnimatedModule(const std::weak_ptr<IReactInstance> & reactInstance);
+    virtual ~NativeAnimatedModule() = default;
+
+    // CxxModule
+    std::string getName() override { return name; };
+    std::map<std::string, folly::dynamic> getConstants() override { return {}; };
+    auto getMethods()->std::vector<Method> override;
+
+    void CreateAnimatedNode(int64_t tag, const folly::dynamic& config);
+    void ConnectAnimatedNodeToView(int64_t animatedNodeTag, int64_t viewTag);
+    void DisconnectAnimatedNodeFromView(int64_t animatedNodeTag, int64_t viewTag);
+    void ConnectAnimatedNodes(int64_t parentNodeTag, int64_t childNodeTag);
+    void DisconnectAnimatedNodes(int64_t parentNodeTag, int64_t childNodeTag);
+    void StartAnimatingNode(int64_t animationId, int64_t animatedNodeTag, const folly::dynamic& animationConfig, const Callback& endCallback);
+    void StopAnimation(int64_t animationId);
+    void DropAnimatedNode(int64_t tag);
+    void SetAnimatedNodeValue(int64_t tag, double value);
+    void SetAnimatedNodeOffset(int64_t tag, double offset);
+    void FlattenAnimatedNodeOffset(int64_t tag);
+    void ExtractAnimatedNodeOffset(int64_t tag);
+    void AddAnimatedEventToView(int64_t tag, const std::string& eventName, const folly::dynamic& eventMapping);
+    void RemoveAnimatedEventFromView(int64_t tag, const std::string& eventName, int64_t animatedValueTag);
+    void StartListeningToAnimatedNodeValue(int64_t tag);
+    void StopListeningToAnimatedNodeValue(int64_t tag);
+
+    static const char* name;
+  private:
+    std::shared_ptr<NativeAnimatedNodeManager> m_nodesManager{};
+    std::weak_ptr<IReactInstance> m_wkReactInstance;
+
+    static constexpr std::string_view s_createAnimatedNodeName{ "createAnimatedNode" };
```

Comment:
whats the advantage of string_view over just a const char*?  we only get the underlying data() ?